File: app/src/main/java/com/autohub/skln/fragment/FragmentHome.java
package com.autohub.skln.fragment;

import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.autohub.skln.R;
import com.autohub.skln.databinding.FragmentTutorHomeBinding;
import com.autohub.skln.utills.GlideApp;
import com.bumptech.glide.load.engine.DiskCacheStrategy;
import com.google.android.gms.tasks.OnFailureListener;
import com.google.android.gms.tasks.OnSuccessListener;
import com.google.firebase.firestore.DocumentSnapshot;
import com.google.firebase.storage.FirebaseStorage;
import com.google.firebase.storage.StorageReference;

import static com.autohub.skln.utills.AppConstants.KEY_FIRST_NAME;

public class FragmentHome extends BaseFragment {
    private FragmentTutorHomeBinding mBinding;

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        return inflater.inflate(R.layout.fragment_tutor_home, container, false);
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        mBinding = FragmentTutorHomeBinding.bind(view);
        setupProfile();
    }

    private void setupProfile() {
        StorageReference ref = FirebaseStorage.getInstance().getReference().child("tutor/" +
                getFirebaseAuth().getCurrentUser().getUid() + ".jpg");
        GlideApp.with(this)
                .load(ref)
                .diskCacheStrategy(DiskCacheStrategy.NONE)  // disable caching of glide
                .skipMemoryCache(true)
                .into(mBinding.ivPicture);

        getFirebaseStore().collection(getString(R.string.db_root_tutors)).document(getFirebaseAuth().getCurrentUser().getUid()).get()
                .addOnSuccessListener(new OnSuccessListener<DocumentSnapshot>() {
                    @Override
                    public void onSuccess(DocumentSnapshot documentSnapshot) {
                        String firstName = documentSnapshot.getString(KEY_FIRST_NAME);
                        mBinding.heyUser.setText("Hey, \n" + firstName + ".");
                    }
                })
                .addOnFailureListener(new OnFailureListener() {
                    @Override
                    public void onFailure(@NonNull Exception e) {
                        showSnackError(e.getMessage());
                    }
                });
    }
}

File: studentmodule/src/main/java/com/autohub/studentmodule/fragments/FragmentHome.kt
package com.autohub.studentmodule.fragments


import android.content.Context
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.LinearLayoutManager
import com.autohub.skln.fragment.BaseFragment
import com.autohub.skln.listeners.ItemClickListener
import com.autohub.skln.models.AcademicData
import com.autohub.skln.models.HobbiesData
import com.autohub.skln.models.UserModel
import com.autohub.skln.utills.ActivityUtils
import com.autohub.skln.utills.GlideApp
import com.autohub.studentmodule.R
import com.autohub.studentmodule.activities.AddClassActivity
import com.autohub.studentmodule.activities.StudentHomeActivity
import com.autohub.studentmodule.adaptors.AcademicAdaptor
import com.autohub.studentmodule.adaptors.HobbiesAdaptor
import com.autohub.studentmodule.databinding.FragmentStudentHomeBinding
import com.autohub.studentmodule.listners.HomeListners
import com.bumptech.glide.load.engine.DiskCacheStrategy

/**
 * Created by Vt Netzwelt
 */
class FragmentHome : BaseFragment() {
    private var mBinding: FragmentStudentHomeBinding? = null
    private var academicAdaptor: AcademicAdaptor? = null
    private var hobbiesAdaptor: HobbiesAdaptor? = null
    private lateinit var homeListner: HomeListners
    private lateinit var user: UserModel
    private val acadmicClickListener = ItemClickListener<AcademicData> {
        homeListner.onAcadmicsSelect(user, it.classname)
    }

    private val hobbiesClickListener = ItemClickListener<HobbiesData> {

    }


    override fun onAttach(context: Context) {
        super.onAttach(context)
        homeListner = context as HomeListners
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        return inflater.inflate(R.layout.fragment_student_home, container, false)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        mBinding = FragmentStudentHomeBinding.bind(view)
        mBinding!!.acadmicrecycleview.layoutManager = LinearLayoutManager(requireContext(), LinearLayoutManager.HORIZONTAL, false)
        academicAdaptor = AcademicAdaptor(requireContext(), acadmicClickListener)
        mBinding!!.acadmicrecycleview.adapter = academicAdaptor

        mBinding!!.hobbiesrecycleview.layoutManager = LinearLayoutManager(requireContext(), LinearLayoutManager.HORIZONTAL, false)
        hobbiesAdaptor = HobbiesAdaptor(requireContext(), hobbiesClickListener)
        mBinding!!.hobbiesrecycleview.adapter = hobbiesAdaptor

        setUpUserInfo()
        mBinding!!.rrGotoclass.setOnClickListener {
            ActivityUtils.launchActivity(requireContext(),
                    AddClassActivity::class.java)
        }

    }

    fun setupProfile() {

        val userimagePath: String? = (context as StudentHomeActivity).userimagePath!!

        if (userimagePath != null && userimagePath != "") {

            GlideApp.with(this)
                    .load(userimagePath)
                    .diskCacheStrategy(DiskCacheStrategy.NONE)  // disable caching of glide
                    .skipMemoryCache(true)
                    .placeholder(com.autohub.skln.R.drawable.default_pic)
                    .into(mBinding!!.profilePicture)
        }

    }

    private fun setUpUserInfo() {
        firebaseStore.collection(getString(R.string.db_root_students)).document(appPreferenceHelper.getuserID()).get()
                .addOnSuccessListener { documentSnapshot ->
                    user = documentSnapshot.toObject(UserModel::class.java)!!

                    (context as StudentHomeActivity).user = user
                    (context as StudentHomeActivity).userimagePath = user.personInfo!!.accountPicture

                    user.id = documentSnapshot.id
                    mBinding!!.heyUser.text = String.format("Hey, \n%s.", user.personInfo!!.firstName)
                    setSubjects(user)
                    setHobbies(user)


                    setupProfile()

                }
                .addOnFailureListener { e -> showSnackError(e.message) }
    }

    /*
    * get student subjects as per 1-10 or 11-12
    *
    * */
    private fun setSubjects(user: UserModel) {
        var grade: String

        firebaseStore.collection("grades").whereEqualTo("id", user.academicInfo!!.selectedGrad).get().addOnSuccessListener {

            it.forEach {
                grade = it.getString("grade")!!

                academicAdaptor!!.setData(user.getAcadmics(grade))
            }
        }


    }

    /*
    * get Student Hobbies
    * */
    private fun setHobbies(user: UserModel) {
        hobbiesAdaptor!!.setData(user.getHobbies())
    }
}
